raiden/tests/smart_contracts/test_netting_channel.py

```diff
@@ -10,1147 +11,1115 @@
-    assert settle_event == {
-        '_event_type': 'ChannelSettled',
-        'block_number': block_number,
-    }
+    with pytest.raises(TransactionFailed):
+        nettingchannel.updateTransfer(transfer0_data, sender=pkey0)
 
-    assert tester_token.balanceOf(address0, sender=privatekey0_raw) == initial_balance0 + deposit - transfer_amount  # noqa
-    assert tester_token.balanceOf(address1, sender=privatekey1_raw) == initial_balance1 + deposit + transfer_amount  # noqa
-    assert tester_token.balanceOf(nettingchannel.address, sender=privatekey1_raw) == 0
 
+def test_update_not_allowed_after_settlement_period(settle_timeout, tester_channels, tester_state):
+    """ updateTransfer cannot be called after the settlement period. """
+    pkey0, pkey1, nettingchannel, channel0, channel1 = tester_channels[0]
 
-def test_close_settle(
-        deposit,
-        settle_timeout,
-        tester_state,
-        tester_channels,
-        tester_events,
-        tester_token):
+    direct0 = make_direct_transfer(channel0, channel1, amount=70, pkey=pkey0)
+    direct0_data = str(direct0.packed().data)
 
-    privatekey0_raw, privatekey1_raw, nettingchannel, channel0, channel1 = tester_channels[0]
-    privatekey0 = PrivateKey(privatekey0_raw, ctx=GLOBAL_CTX, raw=True)
-    privatekey1 = PrivateKey(privatekey1_raw, ctx=GLOBAL_CTX, raw=True)
-    address0 = privatekey_to_address(privatekey0_raw)
-    address1 = privatekey_to_address(privatekey1_raw)
-    unknown_key = tester.k3
+    nettingchannel.close('', sender=pkey0)
+    tester_state.mine(number_of_blocks=settle_timeout + 1)
 
-    initial_balance0 = tester_token.balanceOf(address0, sender=privatekey0_raw)
-    initial_balance1 = tester_token.balanceOf(address1, sender=privatekey1_raw)
+    with pytest.raises(TransactionFailed):
+        nettingchannel.updateTransfer(direct0_data, sender=pkey1)
 
-    transfer_amount0 = 10
-    direct_transfer0 = channel0.create_directtransfer(
-        transfer_amount0,
-        1  # TODO: fill in identifier
-    )
-    direct_transfer0.sign(privatekey0, address0)
 
-    transfer_amount1 = 30
-    direct_transfer1 = channel1.create_directtransfer(
-        transfer_amount1,
-        1  # TODO: fill in identifier
+def test_update_not_allowed_for_the_closing_address(tester_channels):
+    """ Closing address cannot call updateTransfer. """
+    pkey0, pkey1, nettingchannel, channel0, channel1 = tester_channels[0]
+
+    transfer0 = make_direct_transfer(channel0, channel1, amount=10, pkey=pkey0)
+    transfer0_data = str(transfer0.packed().data)
+
+    transfer1 = make_direct_transfer(channel1, channel0, amount=10, pkey=pkey1)
+    transfer1_data = str(transfer1.packed().data)
+
+    nettingchannel.close('', sender=pkey0)
+
+    # do not accept a transfer of it's own
+    with pytest.raises(TransactionFailed):
+        nettingchannel.updateTransfer(transfer0_data, sender=pkey0)
+
+    # nor a transfer from the partner
+    with pytest.raises(TransactionFailed):
```

Comment:
Aha I see. So this tests the failure of the closing party trying to dispute with a transfer of the counterparty.